sycl/test-e2e/CurrentDevice/current_device.cpp

```diff
@@ -0,0 +1,74 @@
+// REQUIRES: cpu, gpu
```

Comment:
This is never `true`.

Reply:
That's still `false`. A device can't be both cpu and gpu at the same time.

Reply:
> This is never `true`.

So you say we have incorrect test for years?

https://github.com/intel/llvm/blob/b97b714c4f629b0f974fd01f2d6953f54c256fb2/sycl/test-e2e/AOT/multiple-devices.cpp#L9

Reply:
That isn't the only one that's wrong.

Reply:
> This is never `true`.

I disagree. It's true when testing environment has both CPU and GPU devices. It's quite common. There are tests checking cooperation between multiple devices.

Having said that, I think CI limits testing environment to a single device, which effectively disables the tests requiring multiple devices in CI.

Reply:
> I disagree.

With what? Both `cpu` and `gpu` "features" are per-device and that **is** the semantics of the markup (in addition to `%{run}` substitution) that we use in our tests. It's a fact that a given device can't be both cpu and gpu at the same time.

As for the environment with both CPU and GPU available, it's also supported in our testing framework, but one has to use the correct features/substitutions to do that. 

https://github.com/intel/llvm/blob/141348135d33cefc6adb73dfa87f20b387e90b19/sycl/test-e2e/lit.cfg.py#L710-L714

and 

https://github.com/intel/llvm/blob/141348135d33cefc6adb73dfa87f20b387e90b19/sycl/test-e2e/format.py#L298

Reply:
`any-device-is-cpu, any-device-is-gpu` doesn't work in my case, I guess I will just remove this e2e test, or revert to `cpu, gpu` as one way or another the feature is tested with the same amount of coverage but via device-agnostic unit tests

Reply:
> or revert to `cpu, gpu`

that would be wrong.

> any-device-is-cpu, any-device-is-gpu doesn't work in my case

Did you switch `%{run}` to `%{run-unfiltered-devices}`?